api/sys_plugins.go

```diff
@@ -217,28 +218,59 @@ type RegisterPluginInput struct {
 	Env []string `json:"env,omitempty"`
 }
 
+type RegisterPluginResponse struct {
+	Warnings []string `json:"warnings"`
+}
+
 // RegisterPlugin wraps RegisterPluginWithContext using context.Background.
-func (c *Sys) RegisterPlugin(i *RegisterPluginInput) error {
+func (c *Sys) RegisterPlugin(i *RegisterPluginInput) (*RegisterPluginResponse, error) {
```

Comment:
Should we add a net-new method on the API so as to not break API consumers, a la TFVP etc?